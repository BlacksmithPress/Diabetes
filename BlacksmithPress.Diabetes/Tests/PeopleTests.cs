using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BlacksmithPress.Diabetes.Cloud.Controllers;
using BlacksmithPress.Diabetes.Data;
using MongoDB.Driver;
using NUnit.Framework;
using ObjectApproval;
using Shouldly;

namespace Tests
{
    public class PeopleTests
    {
        [Test]
        public void PeopleController_Post_CreatesPerson()
        {
            var context = new Context("DefaultConnection");
            var controller = new PeopleController(context);

            var name = "TEST: Joshua";

            if (context.People.Any(p => p.Name == name))
                context.People.RemoveRange(context.People.Where(p => p.Name == name));

            var person = new Person {Name = name};
            var result = controller.PostPerson(person);

            person.Id.ShouldBeGreaterThan(0);

            context.People.RemoveRange(context.People.Where(p => p.Name == name));
            context.SaveChanges();
        }


        [Test]
        public void PeopleController_Put_EditsExistingPerson()
        {
            // arrange isolation
            var context = new Context("DefaultConnection");
            var controller = new PeopleController(context);

            // arrange test
            var actual = context.People.FirstOrDefault();
            if (actual == null)
            {
                actual = new Person {Name = "Pre-existing Person"};
                context.People.Add(actual);
                context.SaveChanges();
            }

            // act
            actual.Name += " (edited)";
            var result = controller.PutPerson(actual.Id, actual);

            // assert
            var expected = context.People.FirstOrDefault(p => p.Id == actual.Id);
            actual.ShouldBe(expected);

            // clean-up
            context.People.Remove(actual);
            context.SaveChanges();
        }

        [Test]
        public void PeopleController_Patch_EditsExistingPerson()
        {
            // arrange isolation
            var context = new Context("DefaultConnection");
            var controller = new PeopleController();

            // arrange test
            var subject = context.People.FirstOrDefault();
            if (subject == null)
            {
                subject = new Person { Name = "Pre-existing Person" };
                context.People.Add(subject);
                context.SaveChanges();
            }
            var id = subject.Id;

            // act
            var actual = subject.Clone();
            actual.Id = default(long);
            actual.Name += " (edited)";
            var result = controller.PatchPerson(id, actual);

            // assert
            context = new Context("DefaultConnection");
            var expected = context.People.FirstOrDefault(p => p.Id == id);
            actual.Name.ShouldBe(expected.Name);

            // clean-up
            context.People.Remove(expected);
            context.SaveChanges();
        }

    }
}
