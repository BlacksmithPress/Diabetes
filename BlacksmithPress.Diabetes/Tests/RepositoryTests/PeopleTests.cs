using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Text;
using System.Threading.Tasks;
using Autofac;
using BlacksmithPress.Diabetes.Entities;
using BlacksmithPress.Diabetes.Persistence.Repositories;
using BlacksmithPress.Diabetes.Types;
using NUnit.Framework;
using ObjectApproval;
using Shouldly;

namespace RepositoryTests
{
    public class PeopleTests : TestBase
    {
        public PeopleTests()
        {
            repository = new People(new NetworkCredential("ken","Password"));
        }

        private People repository;

        [Test]
        public void PeopleRepository_Create_CreatesNewPerson()
        {
            // arrange isolation

            // arrange test

            // act
            var actual =  repository.Create(new Person {Name = "Ken"});

            // assert
            actual.Name.ShouldBe("Ken");
            actual.Id.ShouldBeGreaterThan(0);

            // clean-up
            repository.Delete(actual.Id);
        }


        [Test]
        public void PeopleRepository_Get_GetsAPerson()
        {
            // arrange isolation

            // arrange test
            var expected = repository.Create(new Person {Name = "Existing Person"});

            // act
            var actual = repository.Get(expected.Id);

            // assert
            actual.ShouldBe(expected);

            // clean-up
            repository.Delete(expected.Id);
        }


        [Test]
        public void PeopleRepository_Update_UpdatesExistingPerson()
        {
            // arrange isolation

            // arrange test
            var existing = repository.Create(new Person {Name = "Existing Person"});
            var expected = new Person {Id = existing.Id, Name = "Existing Person (edited)"};

            // act
            var actual = repository.Update(expected);

            // assert
            actual.ShouldBe(expected);

            // clean-up
            repository.Delete(actual.Id);
        }


        [Test]
        public void PeopleRepository_Delete_DeletesPerson()
        {
            // arrange isolation

            // arrange test
            var existing = repository.Create(new Person { Name = "Existing Person" });

            // act
            repository.Delete(existing.Id);
            var actual = repository.Get(existing.Id);

            // assert
            actual.ShouldBeNull();

            // clean-up
        }


        [Test]
        public void PeopleRepository_GetAll_GetsAllPeople()
        {
            // arrange isolation

            // arrange test

            // act
            var actual = repository.GetAll();

            // assert
            ObjectApprover.VerifyWithJson(actual);

            // clean-up
        }


        [Test]
        public void PeopleRepository_GetOrCreate_GetsEthan()
        {
            // arrange isolation
            
            // arrange test

            // act
            var repository = new People(new NetworkCredential());
            var ethan = repository.GetAll().FirstOrDefault(p => p.Name == "Ethan");

            if (ethan == null)
                ethan = repository.Create(new Person { Name = "Ethan" });

            // assert
            ethan.Name.ShouldBe("Ethan");

            // clean-up
        }


    }
}
