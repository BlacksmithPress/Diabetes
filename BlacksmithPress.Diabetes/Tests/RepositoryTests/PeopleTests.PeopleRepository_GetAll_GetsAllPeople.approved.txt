[
  {
    "Name": "Ken",
    "Id": 1
  },
  {
    "Name": "Ethan",
    "Id": 2
  },
  {
    "Name": "Ken",
    "Id": 38
  },
  {
    "Name": "Ken",
    "Id": 39
  },
  {
    "Name": "Ken",
    "Id": 40
  },
  {
    "Name": "Ken",
    "Id": 41
  },
  {
    "Name": "Sophie",
    "Id": 48
  },
  {
    "Name": "Ethan Andrew",
    "Id": 49
  }
]